### DISCUSSION TITLE
Show browser timezone for fixed time event

### DISCUSSION THREAD

**Discussion by 35233larry** [add2cal/add-to-calendar-button#357] [Ideas / Requests]
I've got an event with people from over the world coming up. I'd like to use the button to help them add the event to their calendar. However, it seems the button always shows the event time for the timezone specified in the settings. Problem: A user who's outside that timezone will see the wrong time.

Is there a solution to this?

**Comment by jekuer** [comment]
Not 100% sure what you are up for.
I guess you are using the date style or a multi date event.
this button shows the time zone of the event only of the user is not in that time zone.
I think the question here is whether we show the event date and it's time zone OR calculate the time the event takes place for the user in his time zone and show this directly.
The latter one would be rather unusual, since events are usually promoted with a specific time, which is never bound to the user (imagine a user reading 9am, then changing time zone, and misses the event).
The mentioned calculation therefore should take place at the users calendar and this is why we need to manage the event with the time zone of the host. That way, the event is somehow fixed, while it is up to the user and his calendar when it takes place for them.

Another example:
Let's imagine you want to make the event look more individual to the user and therefore adapt to the user's time zone.
If this user tweets about the event, he would tell everybody "hey, it is at 10pm" - which than would be wrong to users in another time zone. This, of course, can also happen with the current solution, but I would consider it more stable, as times are not completely changing for everybody.

I could imagine making this configurable, but at the very moment if feels like a rather low prio topic.
The workaround could be to not use the date style, but mention the event date independently from the button; only using the button for the add to calendar action.

**Reply by 35233larry** [reply]
So I'm purely talking about the button style where the event date and time are shown on the button.

Imagine I'm running an online webinar to a global audience. In that case there is no fixed timezone for the event. Which means the event takes place at a different time of the date for every user, depending on their timezone. Could even be a different date. For Larry in Berlin, it's at 3am on the 4th of March. But for Mary in New York it's at 9pm on the 3rd of March. And for the host it Singapore, it's at 10am on the 4th of March.

So I'm the host and I add the button to my page and by default it will use Singapore timezone. But if Larry and Mary visit the page, the button will show 10am Singapore time, which granted is accurate. But it's irrelevant for Larry and Mary. To them the only thing that matters is: When is this event in MY timezone.

Hope this makes sense.

**Reply by jekuer** [reply]
What do you think about the following solution.
Underlying idea is to keep the number of parameters small, since it makes life easier for all of us.

The case is only relevant (or strong) for online events or some global happenings.

We show the date in the browser's time zone...

- if the location is an URL (which would make it an online event already), or a magic string like "Global", "Earth", "World", "worldwide".
- adding a suffix "adjusted to your time zone" (which might be too long) or simply "your time zone" or "current browser time zone".

**Comment by jekuer** [comment]
Small update on that: The latest release came with an update regarding how times are mapped to timezones at the date button.
From the docs regarding the `location` option:

> If it is a URL, the event will be classified as "online event" via the schema.org declaration.
> An online event is not showing up on the date button. Additionally, the time on this button type will then be converted to the user's time zone (will also be done, if location is "Global", "Worldwide", or "Online").
